src/agent/rustjail/src/cgroups/systemd/subsystem/cpu.rs

```diff
@@ -50,7 +50,10 @@ impl Cpu {
         systemd_version: &str,
     ) -> Result<()> {
         if let Some(shares) = cpu_resources.shares {
-            properties.push(("CPUShares", Value::U64(shares)));
+            // Minimum value of CPUShares should be 2, see https://github.com/systemd/systemd/blob/d19434fbf81db04d03c8cffa87821f754a86635b/src/basic/cgroup-util.h#L122
+            if shares >= 2 {
+                properties.push(("CPUShares", Value::U64(shares)));
+            }
```

Comment:
So this is basically ignoring an invalid value received from the runtime... not correcting anything unlike the title seems to suggest. Is it appropriate to continue execution ? FWIW the memory subsystem bails out when receiving an invalid value. See https://github.com/kata-containers/kata-containers/blob/main/src/agent/rustjail/src/cgroups/systemd/subsystem/memory.rs#L42 .

Reply:
On second thought, cgroupsv2 treats 0 as a hint to use the default value. Maybe better to do the same in cgroupsv1 for consistency ? WDYT @jongwu @cmaf ?

Reply:
What's the default value of cpuShares? 100?

Reply:
> What's the default value of cpuShares? 100?

See https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/kernel/sched/sched.h#n409

```
/*
 * A weight of 0 or 1 can cause arithmetics problems.
 * A weight of a cfs_rq is the sum of weights of which entities
 * are queued on this cfs_rq, so a weight of a entity should not be
 * too large, so as the shares value of a task group.
 * (The default weight is 1024 - so there's no practical
 *  limitation from this.)
 */
#define MIN_SHARES		(1UL <<  1)
#define MAX_SHARES		(1UL << 18)
```
You can :
- convert runtime input of 0 to 1024
- bail out if value is not in the MIN_SHARES...MAX_SHARES range


 and the value should be in the 2..262144 range.